src/__tests__/testParsing.py

```diff
@@ -339,6 +339,32 @@
         'out': ''
     }]
 
+allInputTestCases = [
+    {
+        'input': '    ',
+        'match': None
+    }, {
+        'input': ' ',
+        'match': None
+    }, {
+        'input': 'a',
+        'match': 'a'
+    }, {
+        'input': '   a',
+        'match': 'a'
+    }, {
+        'input': 'a    ',
+        'match': 'a'
+    }, {
+        'input': '    foo bar',
+        'match': 'foo bar'
+    }, {
+        'input': 'foo bar    ',
+        'match': 'foo bar'
+    }, {
+        'input': 'foo bar baz',
+        'match': 'foo bar baz'
```

Comment:
i know this would pass, but maybe add a test case with whitespace on both sides, just for completeness

```
input: '   foo bar  ',
match: 'foo bar',
```